### DISCUSSION TITLE
How to set position programmatically?

### DISCUSSION THREAD

**Discussion by akay0214** [nerdyman/react-compare-slider#48] [Q&A] [answered]
Hi,

I'm using compare slider inside the modal window. Every time I open modal to compare images for a new item it renders compare strip at the same position I left it before. How do I reset it or set it programmatically?

Thanks,
Alexander.

**Comment by nerdyman** [comment]
Hey Alexander! 

The component will always respect the `position` prop you pass to it so if you want to reset it when the modal opens you could store the position in a `useState` and change it whenever the modal opens with a `useEffect`.

I'm not sure how your modal is controlled but if it's with something like a `visible` prop you could always set the slider position to be `50` whenever visible is true.

E.g.

```jsx
const [visible, setVisible] = useState(false);
const [position, setPosition] = useState(50);

useEffect(() => {
  // Always reset position when visible is true.
  if (visible) setPosition(50);
}, [visible]);

return <ReactCompareSlider position={position} />
```

Here's a super basic example https://codesandbox.io/s/react-compare-slider-simple-reset-example-vgkdh?file=/src/App.jsx

**Reply by akay0214** [reply]
It works only as an initial value, but after you use slider it doesn't react on anything in position variable. I'll double check my code tomorrow and I'll let you know.

**Comment by nerdyman** [comment, marked as answer]
This has been released in [v3](https://github.com/nerdyman/react-compare-slider/releases/tag/v3.0.0)!

See the [`useReactCompareSliderRef` example](https://react-compare-slider.vercel.app/?path=/story/demos--use-react-compare-slider-ref) for usage.

```bash
npm install react-compare-slider@latest
yarn add react-compare-slider@latest
pnpm install react-compare-slider@latest
```
